### PR TITLE
docs: document setProviderAndWait in README

### LINKED ISSUE #592: Update readme examples to include setProviderAndWait
The SDK has support for `setProviderAndWait` since 1.6.0. It was added in [this PR](https://github.com/open-feature/java-sdk/pull/563). Update the README examples to show how it can be used.

```[tasklist]
### Tasks
- [ ] Update the example in the quick start section
- [ ] Expand the provider's section to show both the async and synchronous options.
```

### PR DESCRIPTION
**Pull request by code4Y**
## This PR
- Updated the `setProviderAndWait` example in the quick start section
- Expand the provider's section to show both the async and synchronous options.

### Related Issues
Update readme examples to include `setProviderAndWait`

Fixes #592 

### Notes
If there are any corrections, please let me know.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -139,12 +139,28 @@ See [here](https://javadoc.io/doc/dev.openfeature/sdk/latest/) for the Javadocs.
...
+#### Synchronous Provider Registration  
+For synchronous provider registration, where you set the provider without waiting for initialization, you can use the `setProvider` method directly:  
+
+```java
+    OpenFeatureAPI api = OpenFeatureAPI.getInstance();
+    api.setProvider(new MyProvider());
 ```
 
+This is a straightforward synchronous operation, suitable when you don't need to wait for the provider to initialize before continuing.   
+
```
Comment (toddbaert) [inline README.md:163]: I don't think this terminology is quite correct (or at least it's confusing). I would instead use the word "blocking". `setProviderAndWait` _blocks_ until the provider is ready, while `setProvider` does not.

File: README.md [hunk 2]
```diff
@@ -211,19 +227,41 @@ Clients can be given a name.
...
+```   
+   
+#### Synchronous Named Provider Configuration   
+
+For synchronous provider registration, where you set the provider without waiting for initialization, you can use the `setProvider` method directly, as shown in the following example:  
+
+```java
+FeatureProvider scopedProvider = new MyProvider();
+
+// registering the default provider synchronously
```
Comment (toddbaert) [inline README.md:255]: I don't think you need to repeat yourself here. Just mention that named providers can be set in a blocking/non-blocking way, and then link to the previous doc you wrote above.

**Review by toddbaert — CHANGES_REQUESTED** [review]
Hey @code4Y , this is a great start!

I left a few comments.

Also note, we use PR titles for semantic commit messages - so I've made a change to the title (this will be the commit message when we merge).

**Comment by code4Y** [comment]
Hey @toddbaert, thank you for your guidance!

I've made the requested changes. Please take a look at them and let me know if there's anything else I need to improve.

**Review by toddbaert — APPROVED** [review]
LGTM!

**Comment by toddbaert** [comment]
@code4Y you can disregard the CI failure. It's not related to your change.

**Pull request merged**